test/pack/pack.c

```diff
@@ -49,6 +49,22 @@ static void swap_segments(uintptr_t * starts, uintptr_t * lengths, int x, int y)
     lengths[y] = tmp;
 }
 
+static void host_only_get_ptr_attr(yaksa_info_t * info)
+{
+    static int count = 0;
+    if ((++count) % 2 == 0) {
```

Comment:
Is this thread-safe?

Reply:
Not really. But the consequence is just less than perfect half by half testing.  The same  mechanism is used in `test/pack/pack-cuda.c`. We can make it atomic (do we have atomic library for the testing?) but maybe it isn't worth the trouble.

Reply:
No, randomness is not desirable in this case.  It's better to make both cases deterministic.  If we don't have an atomic library, we could use a lock or decide on the info hint based on the iteration number.

Reply:
Okay